spring-integration-core/src/main/java/org/springframework/integration/endpoint/ReactiveStreamsConsumer.java

```diff
@@ -89,8 +89,8 @@ public ReactiveStreamsConsumer(MessageChannel inputChannel, Subscriber<Message<?
 		Assert.notNull(subscriber, "'subscriber' must not be null");
 		this.inputChannel = inputChannel;
 
-		if (inputChannel instanceof NullChannel && logger.isWarnEnabled()) {
-			logger.warn("The consuming from the NullChannel does not have any effects: " +
+		if (inputChannel instanceof NullChannel) {
+			logger.warn(() -> "The consuming from the NullChannel does not have any effects: " +
```

Comment:
We don't need `Supplier` wrapper here: the message is just plain string constant.
The target logging system still checks its `enabled` before printing the message.
And yes: that `isWarnEnabled()` was really an overhead 😄

Reply:
Ok